src/components/connectors/Grid/ConnectorCards.tsx

```diff
@@ -0,0 +1,146 @@
+import type { ConnectorWithTagQuery } from 'src/api/types';
+import type { ConnectorCardsProps } from 'src/components/connectors/Grid/types';
+import type { TableState } from 'src/types';
+
+import { useEffect, useMemo, useRef, useState } from 'react';
+
+import { Grid, Paper, Typography } from '@mui/material';
+
+import { useQuery } from '@supabase-cache-helpers/postgrest-swr';
+import { FormattedMessage } from 'react-intl';
+
+import { getConnectors } from 'src/api/connectors';
+import Card from 'src/components/connectors/Grid/cards/Card';
+import ConnectorRequestCard from 'src/components/connectors/Grid/cards/ConnectorRequestCard';
+import Detail from 'src/components/connectors/Grid/cards/Detail';
+import LegacyCard from 'src/components/connectors/Grid/cards/LegacyCard';
+import Logo from 'src/components/connectors/Grid/cards/Logo';
+import Title from 'src/components/connectors/Grid/cards/Title';
+import { intlConfig } from 'src/components/connectors/Grid/shared';
+import ConnectorSkeleton from 'src/components/connectors/Grid/Skeleton';
+import useEntityCreateNavigate from 'src/components/shared/Entity/hooks/useEntityCreateNavigate';
+import { semiTransparentBackground } from 'src/context/Theme';
+import { checkErrorMessage, FAILED_TO_FETCH } from 'src/services/shared';
+import { TableStatuses } from 'src/types';
+import { hasLength } from 'src/utils/misc-utils';
+import {
+    getEmptyTableHeader,
+    getEmptyTableMessage,
+} from 'src/utils/table-utils';
+
+export default function ConnectorCards({
+    condensed,
+    protocol,
+    searchQuery,
+}: ConnectorCardsProps) {
+    const navigateToCreate = useEntityCreateNavigate();
+    const isFiltering = useRef(false);
+
+    const [tableState, setTableState] = useState<TableState>({
+        status: TableStatuses.LOADING,
+    });
+
+    const query = useMemo(() => {
+        return getConnectors(searchQuery, 'asc', protocol);
+    }, [searchQuery, protocol]);
+
+    const { data: selectResponse, isValidating, error } = useQuery(query);
+
+    const selectData = useMemo(() => selectResponse ?? [], [selectResponse]);
+
+    const RequestCard = condensed ? (
+        <div key="connector-tile-request" />
+    ) : (
+        <ConnectorRequestCard key="connector-tile-request" />
+    );
+
+    const primaryCtaClick = (row: ConnectorWithTagQuery) => {
+        navigateToCreate(row.connector_tags[0].protocol, {
+            id: row.connector_tags[0].connector_id,
+            advanceToForm: true,
+            expressWorkflow: condensed,
+        });
```

Comment:
Using `condensed` as the flag to know the workflow seems a bit concerning for future updates.